src/libraries/System.Text.Json/tests/System.Text.Json.Tests/System.Text.Json.Tests.csproj

```diff
@@ -5,6 +5,8 @@
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <!-- SYSLIB0020: JsonSerializerOptions.IgnoreNullValues is obsolete -->
     <NoWarn>$(NoWarn);SYSLIB0020</NoWarn>
+    <!-- SYSLIB0049: JsonSerializerOptions.AddContext is obsolete -->
+    <NoWarn>$(NoWarn);SYSLIB0049</NoWarn>
```

Comment:
Would it make sense to only do this in the test file that is explicitly testing `AddContext()`? That way other tests don't start using it.

Reply:
It's being used in a bunch of places already. Why would using it new tests be a source of concern? In some cases it might be necessary to do that.

Reply:
> It's being used in a bunch of places already.

That's why it is a concern. Those tests are no longer testing like a customer. The API is obsolete, so unless the test is explicitly testing something about AddContext, it should be moved to use the APIs that we expect users to call. That way our tests act closer to our customers.

Reply:
This holds for most test suites that were written before major features were added to the library, they were written using a style and APIs that reflected what was available at the time. While it makes sense to extend test coverage when new features are added, I wouldn't be in favour of updating old tests for the sake of keeping up with best practice, since it probably subtracts from test coverage.